### ISSUE TITLE
Bug: Must allow duplicate event names across different states

### ISSUE THREAD

**Issue by estaylorco** [ifandelse/machina.js#61]
I believe I found a bug.

Consider the following code snippet:

```
'month': {
    _onEnter: function () {
        _self.currentView(this.state);
        if (this.priorState !== _self.kInitialized) {
            _self.publishZoom(_self.kIn);
        }
        this.publishViewChanged();
    },
    '/view/change/months': function (selection) {
        //When transitioning to the months state, we need
        //to set both the month and the year since both are
        //relevant in the months state
        this.setSelectedMonth(selection.selection.month());
        this.setSelectedYear(selection.selection.year());
        this.transition(_self.kMonths);
    },
    '/view/change/years': function (selection) {
        this.setSelectedYear(selection.selection.year());
        this.transition(_self.kYears);
    }
},
'months': {
    _onEnter: function () {
        _self.currentView(this.state);
        _self.publishZoom(this.priorState === _self.kYears
            ? _self.kIn
            : _self.kOut);
        this.publishViewChanged();
    },
    '/view/change/month': function (selection) {
        this.setSelectedMonth(selection.selection.month());
        this.transition(_self.kMonth);
    },
    '/view/change/years': function (selection) {
        this.setSelectedYear(selection.selection.year());
        this.transition(_self.kYears);
    }
},
'years': {
    _onEnter: function () {
        _self.currentView(this.state);
        _self.publishZoom(_self.kOut);
        this.publishViewChanged();
    },
    '/view/change/months': function (selection) {
        this.setSelectedYear(selection.selection.year());
        this.transition(_self.kMonths);
    }
}
```

Notice that certain events repeat, but under different states.  The "/view/change/years" event and the "/view/change/months" event repeat, and those are firing twice, regardless of which state the FSM is in.

It seems to me that, to allow for repeats like this, you would need to create a dictionary of states internally that maps a unique identifier to the event.  For example:

"year:/view/change/months" to "/view/change/months" in the _year_ state
"month:/view/change/months" to "/view/change/months in the the _month_ state

In other words, the event emitted should be formed from the name of the state as well as from the name of the event, and then mapped appropriately.
